### PR TITLE
Integrate JS soy proto support

### PR DESCRIPTION
**Pull request by hochhaus**

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: closure/templates/test/greeter_idom_proto.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  /**
+   * Renders HTML greeting as document body.
+   */
+  greet() {
+    var person = new Person();
+    person.setName(this.name_);
+    idom.patchInner(goog.global.document.body, greeter.greet, {person: person});
+  }
+};
+
```
Comment (jart) [inline closure/templates/test/greeter_idom_proto.js]: Nit: extra blank lines at end of file

File: closure/templates/test/greeter_idom_proto.soy [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// limitations under the License.
+
+{namespace io.bazel.rules.closure.soy.greeter autoescape="strict"}
+
+
+/**
+ * Greets a person.
+ */
+{template .greet}
+  {@param person: io.bazel.rules.closure.soy.Person}
```
Comment (jart) [inline closure/templates/test/greeter_idom_proto.soy]: My mind is blown. I can't believe this is actually happening. We can now have protocol buffers at all layers of the stack. Thank you so much @hochhaus and @mikesamuel for making this possible.

File: closure/templates/test/greeter_idom_proto.js (hunk 1 above)
Comment (hochhaus) [inline closure/templates/test/greeter_idom_proto.js]: Good catch. Fixed.

File: closure/templates/test/greeter_idom_proto.soy (hunk 2 above)
Comment (hochhaus) [inline closure/templates/test/greeter_idom_proto.soy]: It is an exciting day! I'm glad to have offered the little bit of help that I could.

Can https://github.com/google/closure-templates/issues/99 be closed?

**Comment by hochhaus** [comment]
PTAL

**Comment by jart** [comment]
LGTM 
Mr. Jenkins: test this please

**Comment by hochhaus** [comment]
Could you merge this Justine?

**Comment by jart** [comment]
Been swamped with work. Merging now.

**Comment by hochhaus** [comment]
Thanks! Goodluck with work.

**Pull request merged**